localstack/services/stepfunctions/asl/component/state/state_execution/state_map/iteration/iterator/distributed_iterator_worker.py

```diff
@@ -0,0 +1,137 @@
+    Job,
+    JobPool,
+)
+from localstack.services.stepfunctions.asl.eval.contextobject.contex_object import Item, Map
+from localstack.services.stepfunctions.asl.eval.environment import Environment
+from localstack.services.stepfunctions.asl.eval.program_state import (
+    ProgramError,
+    ProgramState,
+    ProgramStopped,
+)
+
+
+class DistributedIteratorWorker(InlineIteratorWorker):
+    _map_run_record: MapRunRecord
+
+    def __init__(
+        self,
+        work_name: str,
+        job_pool: JobPool,
+        env: Environment,
+        parameters: Optional[Parameters],
+        map_run_record: MapRunRecord,
+        item_selector: Optional[ItemSelector],
+    ):
+        super().__init__(
+            work_name=work_name,
+            job_pool=job_pool,
+            env=env,
+            parameters=parameters,
+            item_selector=item_selector,
+        )
+        self._map_run_record = map_run_record
+
+    def _eval_job(self, env: Environment, job: Job) -> None:
+        self._map_run_record.item_counter.total.count()
+        self._map_run_record.item_counter.running.count()
+
+        self._map_run_record.execution_counter.total.count()
+        self._map_run_record.execution_counter.running.count()
+
+        job_output = None
+        try:
+            env.context_object_manager.context_object["Map"] = Map(
+                Item=Item(Index=job.job_index, Value=job.job_input)
+            )
+
+            env.inp = job.job_input
+            env.stack.append(env.inp)
+            self._eval_input(env_frame=env)
+
+            job.job_program.eval(env)
+
+            # TODO: verify behaviour with all of these scenarios.
+            end_program_state: ProgramState = env.program_state()
+            if isinstance(end_program_state, ProgramError):
+                self._map_run_record.execution_counter.failed.count()
+                self._map_run_record.item_counter.failed.count()
+                job_output = None
+            elif isinstance(end_program_state, ProgramStopped):
+                self._map_run_record.execution_counter.aborted.count()
+                self._map_run_record.item_counter.aborted.count()
+            else:
+                self._map_run_record.item_counter.succeeded.count()
+                self._map_run_record.item_counter.results_written.count()
+
+                self._map_run_record.execution_counter.succeeded.count()
+                self._map_run_record.execution_counter.results_written.count()
+                self._map_run_record.execution_counter.running.offset(-1)
+
+                job_output = env.inp
+
+        except EvalTimeoutError:
+            self._map_run_record.item_counter.timed_out.count()
+
+        except FailureEventException:
+            self._map_run_record.item_counter.failed.count()
+
+        except Exception:
+            self._map_run_record.item_counter.failed.count()
```

Comment:
Is this logged somewhere else? If not it would be good if we have a least *some* debug logging here. Always a bit dangerous to do a blank Exception with no logging.

Reply:
Good point thanks! I'll add some debug calls for the distributed ones only, as the other modes are handled upstream. Eventually as we support tolerance and exception control workflows for MapRun these logs should no longer be needed.